src/main/Main.hs

```diff
@@ -845,6 +853,21 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
       getPkgOpts wc pkgs =
           map ("-package-id=" ++) <$> mapM (getPkgId wc) pkgs
 
+      getRunCmd args = do
+          pkgComponents <- liftM (map lpvComponents . Map.elems . lpProject) getLocalPackages
+          let executables = filter isCExe $ concatMap Set.toList pkgComponents
+          let argExe = if not (null args) then find (\x -> x == (CExe $ T.pack $ head args)) executables
```

Comment:
I realize that the `not (null args)` ensures that `head args` is total. However, I still prefer to avoid partial function calls, since they can easily turn into bugs with a refactor. Would it be possible to replace this with a pattern match, e.g.:

```haskell
case args of
  [] -> Nothing
  firstArg:_ -> ...
```